src/generic/grid.cpp

```diff
@@ -10852,6 +10787,77 @@ void wxGrid::ClearSelection()
         m_selection->ClearSelection();
 }
 
+void wxGrid::CopySelection()
+{
+#if wxUSE_CLIPBOARD
+    // Coordinates of the selected block to copy to clipboard.
+    wxGridBlockCoords sel;
+
+    // Check if we have any selected blocks and if we don't
+    // have too many of them.
+    const wxGridBlocks blocks = GetSelectedBlocks();
+    wxGridBlocks::iterator iter = blocks.begin();
+    if (iter == blocks.end())
+    {
+        // No selection, copy just the current cell.
+        if (m_currentCellCoords == wxGridNoCellCoords)
+        {
+            // But we don't even have it -- nothing to do then.
+            return;
+        }
+
+        sel = wxGridBlockCoords(GetGridCursorRow(),
+            GetGridCursorCol(),
+            GetGridCursorRow(),
+            GetGridCursorCol());
+    }
+    else // We do have at least one selected block.
+    {
+        sel = *blocks.begin();
+
+        if (++iter != blocks.end())
+        {
+            // As we use simple text format, we can't copy more
+            // than one block to clipboard.
+            wxLogWarning
+            (
+                _("Copying more than one selected block "
+                    "to clipboard is not supported.")
+            );
+            return;
```

Comment:
I think it's not very helpful for the new function to give this warning, so I'd consider changing the code further to make the new function return `bool` indicating whether anything has been copied or not and giving this message in the original event handler (alternative: make the function return the error message, as string, with empty string meaning "no error", but it looks like an overkill here).

If not, the documentation should at least mention that the function logs a warning if there is more than one selected block.